covalent/_workflow/electron.py

```diff
@@ -238,7 +238,13 @@ def get_item(e, key):
 
                 get_item.__name__ = node_name
 
-                get_item_electron = Electron(function=get_item, metadata=self.metadata.copy())
+                iterable_metadata = self.metadata.copy()
+                iterable_metadata[
```

Comment:
Hi @FyzHsn, unfortunately this approach is a bit too blunt. For example, what would happen if a pip dep is needed to interpret the inputs to `get_item`? See #1191.